src/pip/_internal/utils/misc.py

```diff
@@ -109,16 +108,15 @@ def rmtree_errorhandler(func, path, exc_info):
     """On Windows, the files in .svn are read-only, so when rmtree() tries to
     remove them, an exception is thrown.  We catch that here, remove the
     read-only attribute, and hopefully continue without problems."""
-    # if file type currently read only
-    if os.stat(path).st_mode & stat.S_IREAD:
-        # convert to read/write
-        os.chmod(path, stat.S_IWRITE)
-        # use the original function to repeat the operation
-        func(path)
-        return
-    else:
+    # if file type is not currently read only
+    if not os.stat(path).st_mode & stat.S_IREAD:
         raise
 
+    # convert to read/write
+    os.chmod(path, stat.S_IWRITE)
+    # use the original function to repeat the operation
+    func(path)
```

Comment:
Does there need to be a `return None` here if that's the pattern you wanted to follow?

Reply:
I've followed that pattern for cases where we're returning None (as a value) vs just the control flow.

pylint counts them as `inconsistent-return-statements`